### DISCUSSION TITLE
Reasoning behind explicitly defining identical VMA and LMA

### DISCUSSION THREAD

**Discussion by amir-skn** [ARM-software/CMSIS_6#144] [General]
Hi,

I am little bit confused by this bit here:
https://github.com/ARM-software/CMSIS_6/blame/65fe25f43bd0d0531baa4540533e445becb6d5b8/CMSIS/Core/Template/Device_M/Config/Device_gcc.ld#L263
Is there any reason that we define indentical LMA and VMA separately for bss while it can just be:
`} > RAM`

In [GNU linker script manual](https://ftp.gnu.org/old-gnu/Manuals/ld-2.9.1/html_chapter/ld_toc.html#TOC21), it says:
"The default (if you do not use the AT keyword) is to make the load address the same as the relocation address"

Many thanks
Amir

**Comment by JonatanAntoni** [comment]
Hi @amir-skn,

This has been added at some point to work around issues in specific tool combinations. We observed different ELF attributes being set by linkers and requested/required/used by tools (such as Fast Models or Arm Debugger). There have been situations where a resulting ELF file was not correctly loaded depending on the attributes in the file. By overspecifying the attributes in the linker script we could get this fixed. Thus, we decided to add this to the templates in order to achieve a wide compatibility. Ultimately, it's up to the user to adopt the linker script to the project needs.

Cheers,
Jonatan
